torchvision/datasets/__init__.py

```diff
@@ -17,6 +17,7 @@
 from .caltech import Caltech101, Caltech256
 from .celeba import CelebA
 from .sbd import SBDataset
+from .vision import VisionDataset
 
 __all__ = ('LSUN', 'LSUNClass',
```

Comment:
Shouldn't `VisionDataset` contained in `__all__` if we want to expose it?

Reply:
If we want it to be imported via `from torchvision.datasets import *`, then yes.

Reply:
Let me rephrase: Do we want that? An _all or nothing_ approach seems reasonable to me.

Reply:
I'm fine with either, but for consistency let's add it there.

Reply:
In my case, I wanted to restrict the folders that are found by mucking with `_find_classes` but the easiest way to do that was just to inherit from VisionDataset and write my own `DatasetFolder` class.
If you want, I can write a pull to allow a keyword arg to ImageFolder to only allow it to find certain folders, which is actually what I need....
not really using this code for what it was intended to do so not sure if you really want that.
thanks for approving

Reply:
I can merge this as is, but I'd accept a PR adding `VisionDataset` to `__all__`